test/test_solvers.jl

```diff
@@ -296,22 +297,169 @@ function can_solve_batched_tridiagonal_system_with_3D_functions(arch, Nx, Ny, Nz
     return Array(ϕ) ≈ ϕ_correct
 end
 
+function vertically_stretched_grid_div_free_velocity(arch, Nx, Ny, zF)
+    Lx, Ly, Lz = 1, 1, zF[end]
+    Δx, Δy = Lx/Nx, Ly/Ny
+
+    #####
+    ##### Vertically stretched operators
+    #####
+
+    @inline δx_caa(i, j, k, f) = @inbounds f[i+1, j, k] - f[i, j, k]
+    @inline δy_aca(i, j, k, f) = @inbounds f[i, j+1, k] - f[i, j, k]
+    @inline δz_aac(i, j, k, f) = @inbounds f[i, j, k+1] - f[i, j, k]
+
+    @inline ∂x_caa(i, j, k, Δx,  f) = δx_caa(i, j, k, f) / Δx
+    @inline ∂y_aca(i, j, k, Δy,  f) = δy_aca(i, j, k, f) / Δy
+    @inline ∂z_aac(i, j, k, ΔzF, f) = δz_aac(i, j, k, f) / ΔzF[k]
+
+    @inline ∂x²(i, j, k, Δx, f)       = (∂x_caa(i, j, k, Δx, f)  - ∂x_caa(i-1, j, k, Δx, f))  / Δx
+    @inline ∂y²(i, j, k, Δy, f)       = (∂y_aca(i, j, k, Δy, f)  - ∂y_aca(i, j-1, k, Δy, f))  / Δy
+    @inline ∂z²(i, j, k, ΔzF, ΔzC, f) = (∂z_aac(i, j, k, ΔzF, f) - ∂z_aac(i, j, k-1, ΔzF, f)) / ΔzC[k]
+
+    @inline div_ccc(i, j, k, Δx, Δy, ΔzF, u, v, w) = ∂x_caa(i, j, k, Δx, u) + ∂y_aca(i, j, k, Δy, v) + ∂z_aac(i, j, k, ΔzF, w)
+
+    @inline ∇²(i, j, k, Δx, Δy, ΔzF, ΔzC, f) = ∂x²(i, j, k, Δx, f) + ∂y²(i, j, k, Δy, f) + ∂z²(i, j, k, ΔzF, ΔzC, f)
+
+    #####
+    ##### Generate "fake" vertically stretched grid
```

Comment:
It's just fake because I'm actually trying to solve for pressure on a vertically stretched grid but create and pass a `RegularCartesianGrid` to the solver.